ulimit.go

```diff
@@ -103,6 +104,13 @@ func ParseUlimit(val string) (*Ulimit, error) {
 		if soft > *hard {
 			return nil, fmt.Errorf("ulimit soft limit must be less than or equal to hard limit: %d > %d", soft, *hard)
 		}
+		var rlimit syscall.Rlimit
+		if err := syscall.Getrlimit(limitType, &rlimit); err == nil {
+			if *hard > int64(rlimit.Max) {
+				return nil, fmt.Errorf("ulimit hard limit must be less than or equal to hard limit for the current process, hard: %d", *hard)
```

Comment:
For debugging purposes, it could be helpful to print `rlimit.Max` as well.
```golang
return nil, fmt.Errorf("ulimit hard limit (%d) must be less than or equal to hard limit for the current process (%d)", *hard, rlimit.Max)
```